lib/hexpm/repository/release_retirement.ex

```diff
@@ -8,7 +8,7 @@ defmodule Hexpm.Repository.ReleaseRetirement do
     field :message, :string
   end
 
-  @reasons ~w(other invalid security deprecated renamed)
+  @reasons ~w(other invalid security deprecated renamed report)
```

Comment:
This reason should only be able to be set internally, users should not be able to set it when manually retiring with the task `mix hex.retire`. You can pass down an option through `Releases.retire` that either enable or disable this reason. Another option is to use a separate a changeset when retiring because of a package report.